packages/material-ui/src/Stack/Stack.js

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {
+  createUnarySpacing,
+  getValue,
+  handleBreakpoints,
+  unstable_extendSxProp as extendSxProp,
+} from '@material-ui/system';
+import { deepmerge } from '@material-ui/utils';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
+
+/**
+ * Return an array with the separator React element interspersed between
+ * each React node of the input children.
+ *
+ * > intersperse([1,2,3], 0)
+ * [1,0,2,0,3]
+ */
+function intersperse(children, separator) {
```

Comment:
I wonder if we shouldn't make this module public.

Reply:
I think so 👍  how do you feel about renaming to `joinChildren`, similar to `Array.prototype.join`?

Reply:
`joinChildren` sounds like a better name.